### DISCUSSION TITLE
Trim spaces from content entered by user during saving entries

### DISCUSSION THREAD

**Discussion by robsonsobral** [ExpressionEngine/ExpressionEngine#3266] [Feature Requests]
This is something that intrigues me.

Years ago, I had a client who used to add a lot of line breaks at the end of long text fields. Now, I have one who creates entries with white spaces at the beginning of titles! Why? I've no idea. I guess she copies and paste, bringing the white spaces.

So, why not to trim text fields of content?

**Comment by eric-lamb62** [comment]
I can't speak for EE, but for me, when I build a thing with user input being as a part of it, I tend to put the burden on the user to know what they're doing (unless otherwise directed). Basically, the program can't stand in the way of the user. What they input is what should be output. Safely and sanely, but still, "apples === apples" always.

**Comment by shbchk** [comment]
Yes, that is super annoying! I guess, we need some option to trim whitespaces - on input stage or output or both.

**Comment by eric-lamb62** [comment]
maybe, a ":trim", ":rtrim", and ":ltrim" modifier?

**Comment by robsonsobral** [comment]
There's already a `:trim` modifier coming soon.

I get the point of respect whatever the user submitted, however... This looks to me as bad validation. WTH to have spaces at the beginning of a title?!

I guess this could be a field config.

**Comment by shbchk** [comment]
:trim modifier is definitely good, but what if I am already using modifier, :attr_safe, for example, can I use two modifiers at the same time?

**Comment by eric-lamb62** [comment]
I'd be pretty disappointed if you couldn't chain modifiers, but I'm only guessing. You know if that's possible @robsonsobral?

**Comment by robsonsobral** [comment]
I never tried, @mithra62. Let's do it!

No, the modifiers can't be chained. Even worst, they are parsed from right to left (like CSS selectors), which means only the last one is used, which I found weird.

I've tested using:

```
{title}

{title:url_slug limit="10" remove_stopwords="yes" plus_encoded_spaces="yes" find="a" replace="4"}
{title:replace:url_slug limit="10" remove_stopwords="yes" plus_encoded_spaces="yes" find="a" replace="4"}
{title:url_slug:replace limit="10" remove_stopwords="yes" plus_encoded_spaces="yes" find="a" replace="4"}
{title:url_slug:url_encode limit="10" remove_stopwords="yes" plus_encoded_spaces="yes" find="a" replace="4"}
{title:url_encode:url_slug limit="10" remove_stopwords="yes" plus_encoded_spaces="yes" find="a" replace="4"}
```

Which gave me:

```
Segunda via de CRLV

via-crlv
via-crlv
Segund4 vi4 de CRLV
Segunda+via+de+CRLV
via-crlv                                    
```

**Comment by eric-lamb62** [comment]
Welp... okie... dokie... then... O.o

Must be a reason, but gonna take a look at the internals and see if there's a reason for that and/or what the downside is to allow for chaining. 

FWIW though, chaining aside, my main concern is to trim by default. Make it available, as an option, and I'm all in. Just not by default please (for the love of God please).

**Comment by robsonsobral** [comment]
Sure! We can't make a breaking change by default!

**Comment by shbchk** [comment]
Any news on modifiers chaining?

**Comment by intoeetive** [comment]
@shbchk Nope, not yet (the issue for that is https://github.com/ExpressionEngine/ExpressionEngine/issues/136)

**Comment by intoeetive** [comment]
`:trim` modifier has been added and the support for modifiers chaining is coming in 7.3.0

Closing this issue

**Comment by robsonsobral** [comment]
@intoeetive , this issue was about to trim values on field filling. Is still possible to save a title as `              my title has spaces`?

**Comment by intoeetive** [comment]
@robsonsobral sorry, misunderstood it!

The content that editors enter in the entry remains just the way it is. EE never modifies user content, unless there's XSS threat. So yes, the spaces will be preserved.

I can see WHY you (and other people) might want different behavior though, so reopening and converting this to discussion

**Comment by robsonsobral** [comment]
Besides titles, metadata of files should be trimmed:

- Title
- Description
- Credit
- Location

I can't see a reason to preserve white spaces.

**Comment by swierczek** [comment]
+1 to this. And adding a config option to any text-based fieldtype would be the best of both worlds - by default, it wouldn't trim, but it would allow admins to toggle this for any fields as needed.

Example:
<img width="619" alt="image" src="https://github.com/user-attachments/assets/253a76c6-7e68-4774-b1ee-0d22d08f3eca" />

**Reply by robsonsobral** [reply]
I think it could be enabled by default on new installations, but... That's just an idea.
